tests/Operation/WatchFunctionalTest.php

```diff
@@ -380,6 +377,8 @@ private function assertStartAtOperationTime(TimestampInterface $expectedOperatio
 
     public function testRewindMultipleTimesWithResults()
     {
+        $this->skipIfIsShardedCluster('Cursor needs multiple getMore commands prohibiting rewinding it.');
```

Comment:
Can this be rephrased? It's not clear to me why we're actually skipping this test. In particular, "commands prohibiting rewinding it" is difficult to parse.

I assume the conflict is that the test expects a single call to `next()` to advance to a valid position, and that doesn't fly with sharded clusters. But if `rewind()` is always a NOP, why can't we change the first `next()` call to `advanceCursorUntilValid()`? Wouldn't that allow up to five attempts to call `next()` and return if we reach a valid position (i.e. first result document)? The subsequent `next()` could also use `advanceCursorUntilValid()`, but I think we could also rely on `next()` _not_ to invoke a `getMore` there since both result documents should have been returned in the same batch.

Reply:
I'll reword the skip message. Using `advanceCursorUntilValid` causes the internal cursor to throw an exception once we call `rewind`, as it has advanced already.

We may be able to fix this by changing the `isRewindNop` functionality in `ChangeStreamIterator`, but the underlying cursor definitely counts as advanced, prohibiting a `rewind` after multiple calls to `next` on the change stream. I'll investigate this separately to get to the bottom of it